pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -384,8 +384,8 @@ public ClientConfigurationData getConfiguration() {
             // gets the next single threaded executor from the list of executors
             ExecutorService listenerThread = externalExecutorProvider.getExecutor();
             if (metadata.partitions > 1) {
-                consumer = new PartitionedConsumerImpl<>(PulsarClientImpl.this, conf, metadata.partitions, listenerThread,
-                        consumerSubscribedFuture, schema);
+                consumer = new TopicsConsumerImpl<>(PulsarClientImpl.this, conf,
```

Comment:
How does `TopicsConsumerImpl` discover the partitions it has to use?

Reply:
Thanks, this is similar to what we did in partitioned consumer, mainly by `getPartitionedTopicMetadata`.
In `TopicsConsumerImpl` we call `subscribeAsync(topic)` for the topic in conf, and this will call `getPartitionedTopicMetadata` again.

Reply:
Ok, though since we already know the number of partitions, we already know the list of partition names. Couldn't we just build the list and pass it on to the `MultiTopicsConsumerImpl`?

Reply:
Thanks, will do the change to avoid the dup calling of get metadata.